samples/drivers/rtc/ds1307/prj.conf

```diff
@@ -0,0 +1,12 @@
+# Enable Peripheral
+CONFIG_I2C=y
+CONFIG_RTC=y
+CONFIG_RTC_DS1307=y
+
+# Enable SHELL
+CONFIG_SHELL=n
+CONFIG_I2C_SHELL=n
+CONFIG_RTC_SHELL=n
```

Comment:
The default for these options is already `=n`, so these can be removed

Reply:
I removed all unnecessary Kconfig options and changed the overlay as you suggested.